python/ray/tune/ray_trial_executor.py

```diff
@@ -210,16 +218,17 @@ def start_trial(self, trial, checkpoint=None):
             checkpoint (Checkpoint): A Python object or path storing the state
                 of trial.
         """
+        attempts = trial.num_failures_since_result
```

Comment:
why not just use `num_failures`?

Reply:
oh because a trial can fail at multiple times in its lifetime. imagine a scenario where it fails twice, but recovers on 3rd attempt, then fails again for another reason. in that case you want the attempts to be reset.

Reply:
but this introduces two different levels of failure counting right? why not just keep it with 1?

Reply:
both are necessary but i agree the semantics are confusing let me try to simplify it somehow